api/src/main/java/org/openmrs/module/clientregistry/api/event/PatientCreateUpdateListener.java

```diff
@@ -0,0 +1,99 @@
+package org.openmrs.module.clientregistry.api.event;
+
+import javax.jms.JMSException;
+import javax.jms.MapMessage;
+import javax.jms.Message;
+
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.hl7.fhir.r4.model.HumanName;
+import org.hl7.fhir.r4.model.Identifier;
+import org.hl7.fhir.r4.model.Patient;
+import org.openmrs.api.context.Daemon;
+import org.openmrs.event.EventListener;
+import org.openmrs.module.DaemonToken;
+import org.openmrs.module.clientregistry.ClientRegistryConfig;
+import org.openmrs.module.clientregistry.ClientRegistryConstants;
+import org.openmrs.module.fhir2.api.FhirPatientService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PatientCreateUpdateListener implements EventListener {
+	
+	private Log log = LogFactory.getLog(this.getClass());
+	
+	private DaemonToken daemonToken;
+	
+	@Autowired
+	private FhirPatientService patientService;
+	
+	@Autowired
+	private ClientRegistryConfig config;
+	
+	@Autowired
+	private IGenericClient client;
+	
+	public DaemonToken getDaemonToken() {
+		return daemonToken;
+	}
+	
+	public void setDaemonToken(DaemonToken daemonToken) {
+		this.daemonToken = daemonToken;
+	}
+	
+	@Override
+	public void onMessage(Message message) {
+		log.trace("Received message: " + message);
+
+		Daemon.runInDaemonThread(() -> {
```

Comment:
also curious if there's any recovery we can do if `runInDaemonThread` throws an exception (the method itself isn't defined throwing any exceptions, but i'm pretty sure it can throw them)